code/acting/src/acting/DummyTrajectorySub.py

```diff
@@ -33,12 +48,76 @@ def __init__(self):
         self.target_wp = (100.0, 0.0)
         self.target_trajectory = [self.start_wp, self.target_wp]
 
+        # Subscriber
         self.dummy_trajectory_subscriber = self.new_subscription(
             Path,
             "/carla/" + self.role_name + "/trajectory",
             self.update_trajectory,
             qos_profile=1)
 
+        self.pos_counter = 0
+        self.pos_average = [0, 0, 0]
+        self.gnss_subscriber = self.new_subscription(
+            NavSatFix,
+            "/carla/" + self.role_name + "/GPS",
+            self.output_average_gps_2_xyz,
+            qos_profile=1)
+
+        # Publisher
+        self.pos_publisher = self.new_publisher(
+            PoseStamped,
+            "/carla/" + self.role_name + "/current_pos",
+            qos_profile=1)
+
+    def output_gps_2_xyz(self, data: NavSatFix):
+        """
+        Transforms GPS coordinates into local coordinates
+        using the CoordinateTransformer transformer.
+        The current position is then updated and published
+        in the PoseStamped format.
+        :param data: message according to NavSatFix definition
+        :return:
+        """
+        if self.pos_counter % 4 == 0:
+            lat = data.latitude
+            lon = data.longitude
+            h = data.altitude
+            x, y, z = self.transformer.gnss_to_xyz(self, lat, lon, h)
+            self.update_pose(x, y, z)
+            # self.loginfo(f"x: {x}\t y: {y}\t z:{z}")
+
+        self.pos_counter += 1
+
+    def output_average_gps_2_xyz(self, data: NavSatFix):
```

Comment:
Is this average better than the individual values?

Reply:
I used the average, because the noise in the GPS sensor is big enough to influence the steering angle quite a lot.
However, noise is a topic that should be discussed, as the currently the pure pursuit controller doesn't work due to the sensor signal. In later stages we could use the signal directly, that's why there is a second function that is not using the average.